qiime/core/archive/format/util.py

```diff
@@ -22,3 +28,64 @@ def artifact_version(version):
         yield
     finally:
         Archiver.CURRENT_FORMAT_VERSION = original_version
+
+
+# parse_actions is a private utility helper that accepts v0- and v1- format
+# Artifacts, and returns a NetworkX DiGraph where the nodes represent artifacts
+# and the edges represent actions. The nodes and edges each carry the necessary
+# attributes (as extracted from the provenance) to recreate the execution
+# pipeline necessary to derive the  artifact being processed.
+# This utility is intended to be a stop-gap measure for now, and we anticipate
+# replacing it with a more full-featured provenance object.
+def parse_actions(artifact):
+    prov_dir = artifact._archiver.provenance_dir
+    prov = nx.DiGraph()
+
+    # v0-format artifact, nothing we can do here.
+    if prov_dir is None or not prov_dir.exists():
+        prov.add_node(artifact.uuid)
+        return prov
+
+    def dict_list_to_ordereddict(l):
+        return OrderedDict([tuple(x.items())[0] for x in l])
+
+    def load_yaml(prov_dir):
+        action_yaml = prov_dir / 'action' / 'action.yaml'
+        metadata_yaml = prov_dir / 'metadata.yaml'
```

Comment:
`ProvenanceCapture` has some of these constants, but maybe they aren't quite as useful where they are? These details strike me as specific to the format version, so some refactoring may be needed in the future.